try {} catch (e) { using x = resource; }